File: api/src/main/java/dev/ulloasp/mlsuite/organization/application/dto/OrganizationDto.java
package dev.ulloasp.mlsuite.organization.application.dto;

import java.time.OffsetDateTime;

import dev.ulloasp.mlsuite.organization.domain.model.Organization;

public record OrganizationDto(
        Long id,
        String slug,
        String name,
        String description,
        String avatarUrl,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt) {

    public static OrganizationDto from(Organization organization) {
        return new OrganizationDto(
                organization.getId(),
                organization.getSlug(),
                organization.getName(),
                organization.getDescription(),
                organization.getAvatarUrl(),
                organization.getCreatedAt(),
                organization.getUpdatedAt());
    }
}

File: frontend/src/api/workspace/dtos/organization-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/


export interface OrganizationDto {
  id: number;
  slug: string;
  name: string;
  description?: string | null;
  avatarUrl?: string | null;
  createdAt: string;
  updatedAt: string;
}
